packages/datatrak-web-server/src/routes/TasksRoute.ts

```diff
@@ -41,79 +40,128 @@ type SingleTask = Task & {
+    const cookieString = this.req.headers.cookie;
+    if (!cookieString) {
+      return;
+    }
+    const cookies = getFilterSettings(cookieString);
 
-  return models.database.count(RECORDS.TASK, filtersWithColumnSelectors, {
-    multiJoin: models.task.DatabaseRecordClass.joins,
-  });
-};
+    if (!cookies.allAssignees) {
+      const { id: userId } = await this.req.ctx.services.central.getUser();
+      this.filters['assignee_id'] = userId;
+    }
 
-const EQUALITY_FILTERS = ['due_date', 'survey.project_id', 'task_status'];
+    // If the task status filter is already present, don't need to worry about allCompleted and allCancelled filters
+    if ('task_status' in this.filters) {
+      return;
+    }
 
-const formatFilters = (filters: Record<string, string>[]) => {
-  let formattedFilters: FormattedFilters = {};
+    if (!cookies.allCompleted) {
+      this.filters['task_status'] = {
+        comparator: 'NOT IN',
+        comparisonValue: [TaskStatus.completed],
+      };
+    }
 
-  filters.forEach(({ id, value }) => {
-    if (value === '' || value === undefined || value === null) return;
-    if (EQUALITY_FILTERS.includes(id)) {
-      formattedFilters[id] = value;
-      return;
+    if (!cookies.allCancelled) {
+      this.filters['task_status'] = {
+        comparator: 'NOT IN',
+        comparisonValue: [TaskStatus.cancelled],
+      };
     }
 
-    if (id === 'repeat_schedule') {
-      formattedFilters[id] = {
-        comparator: 'ilike',
-        comparisonValue: `${value}%`,
-        castAs: 'text',
+    if (!cookies.allCompleted && !cookies.allCancelled) {
+      this.filters['task_status'] = {
+        comparator: 'NOT IN',
+        comparisonValue: [TaskStatus.completed, TaskStatus.cancelled],
       };
-      return;
     }
-    formattedFilters[id] = {
-      comparator: 'ilike',
-      comparisonValue: `${value}%`,
-    };
-  });
+  }
+
+  private async queryForCount() {
+    const { models } = this.req;
+    const filtersWithColumnSelectors = { ...this.filters };
+
+    // use column selectors for custom columns being used in filters
+    for (const [key, value] of Object.entries(this.filters)) {
+      if (key in models.task.customColumnSelectors) {
+        const colKey =
+          models.task.customColumnSelectors[
+            key as keyof typeof models.task.customColumnSelectors
+          ]();
+        filtersWithColumnSelectors[colKey] = value;
+        delete filtersWithColumnSelectors[key];
+      }
+    }
+
+    return models.database.count(RECORDS.TASK, filtersWithColumnSelectors, {
+      multiJoin: models.task.DatabaseRecordClass.joins,
+    });
+  }
 
-  return formattedFilters;
-};
-export class TasksRoute extends Route<TasksRequest> {
   public async buildResponse() {
     const { ctx, query = {}, models } = this.req;
-    const { filters = [], pageSize = DEFAULT_PAGE_SIZE, sort, page = 0 } = query;
+    const { pageSize = DEFAULT_PAGE_SIZE, sort, page = 0 } = query;
+
+    this.formatFilters();
```

Comment:
I refactored the route to use class methods instead of pure functions for the filtering logic because I thought it was cleaner than passing around the req properties such as model. It also meant that they could just access a consistent filter property which has a predictable type in this.filters.